flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -780,25 +783,32 @@ private static void setManagedMemoryFractionForOperator(
 		if (groupResourceSpec.equals(ResourceSpec.UNKNOWN)) {
 			checkArgument(groupOperatorCount > 0, "A slot sharing group must contain at least 1 operator");
 
-			managedMemoryFractionOnHeap = 1.0 / groupOperatorCount;
-			managedMemoryFractionOffHeap = 1.0 / groupOperatorCount;
+			final double fraction = getFractionRoundedDown(1, groupOperatorCount);
+			managedMemoryFractionOnHeap = fraction;
+			managedMemoryFractionOffHeap = fraction;
 		} else {
 			final long groupOnHeapManagedMemoryBytes = groupResourceSpec.getOnHeapManagedMemory().getBytes();
 			final long groupOffHeapManagedMemoryBytes = groupResourceSpec.getOffHeapManagedMemory().getBytes();
 
 			managedMemoryFractionOnHeap = groupOnHeapManagedMemoryBytes > 0
-				? (double) operatorResourceSpec.getOnHeapManagedMemory().getBytes() / groupOnHeapManagedMemoryBytes
+				? getFractionRoundedDown(operatorResourceSpec.getOnHeapManagedMemory().getBytes(), groupOnHeapManagedMemoryBytes)
 				: 0.0;
 
 			managedMemoryFractionOffHeap = groupOffHeapManagedMemoryBytes > 0
-				? (double) operatorResourceSpec.getOffHeapManagedMemory().getBytes() / groupOffHeapManagedMemoryBytes
+				? getFractionRoundedDown(operatorResourceSpec.getOffHeapManagedMemory().getBytes(), groupOffHeapManagedMemoryBytes)
 				: 0.0;
 		}
 
 		operatorConfig.setManagedMemoryFractionOnHeap(managedMemoryFractionOnHeap);
 		operatorConfig.setManagedMemoryFractionOffHeap(managedMemoryFractionOffHeap);
 	}
 
+	private static double getFractionRoundedDown(final long value, final long totalValue) {
```

Comment:
I see no reason to not name the variables `dividend` and `divisor`

Reply:
Good suggestion!